src/jobflow_remote/remote/data.py

```diff
@@ -165,4 +183,15 @@ def resolve_job_dict_args(job_dict: dict, store: JobStore) -> dict:
     # substitution is in place
     job_dict["function_args"] = resolved_args
     job_dict["function_kwargs"] = resolved_kwargs
+
+    expected_fields = set(JobWrapper.__annotations__)
+    additional_store_names = set(job_dict.keys()) - expected_fields
+    for additional_store in additional_store_names:
+        if additional_store.startswith("@"):
+            continue
+        if additional_store not in store.additional_stores:
+            raise ValueError(
+                f"Additional store {additional_store!r} is not configured."
```

Comment:
I would raise a `RemoteError` with `no_retry=True` here. This would avoid that the runner retries multiple times. See also https://github.com/Matgenix/jobflow-remote/issues/76.